### PR TITLE
tutorial for subproc strings

### LINKED ISSUE #621: quote removal in subprocess mode does not behave as expected
Please have a look at the following two examples::

```
    xonsh$ /bin/echo 'foo'
    foo

    xonsh$ /bin/echo 'foo'bar
    'foo'bar
```

In the first case quotes are stripped from the second argument; in the
second, quotes are kept.

This is not consistent with what other shells do, and makes it quite
inconvenient to pass option arguments containing spaces.  For example,
the following would include the quotes in the commit message::

```
    svn commit --message='A short description here.'
```

### PR DESCRIPTION
**Pull request by scopatz**
Hopefully this closes #621 and #1432

### CODE REVIEW

**Comment by scopatz** [comment]
I don't think this needs a news entry

File: docs/tutorial_subproc_strings.rst [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    $ echo foo"bar"baz
+    foobarbaz
+
+    $ echo --key="value"
+    --key=value
+
+Xonsh considers this behaviour insane. Instead, xonsh treats these
+arguments as if they were surrounded in another, outer level of
+quotation (``'foo"bar"baz'``). Xonsh will keep the quotation marks
+when leading and triling quotes are not matched.
```
Comment (melund) [inline docs/tutorial_subproc_strings.rst]: trailing?

Comment (scopatz) [reply docs/tutorial_subproc_strings.rst]: Thanks for spotting this!

**Comment by scopatz** [comment]
Thanks @melund! I have fixed the typo

File: docs/tutorial_subproc_strings.rst [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+1. Strings in xonsh are always parsed in the same way, and
+2. Python always wins!
+
+Together these rules mean that **even strings in subprocess mode are treated
+like Python strings!** This will (help) preserve your sanity.
+
+No Escape
+=========
+Xonsh strings are exactly like Python strings everywhere. Xonsh uses
+exactly the same escape characters that Python dones; no more and no less.
```
Comment (gforsyth) [inline docs/tutorial_subproc_strings.rst]: *dones -> does

File: docs/tutorial_subproc_strings.rst [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+.. code-block:: bash
+
+    $ echo foo"bar"baz
+    foobarbaz
+
+    $ echo --key="value"
+    --key=value
+
+Xonsh considers this behaviour insane. Instead, xonsh treats these
```
Comment (gforsyth) [inline docs/tutorial_subproc_strings.rst]: Pedantic, but if we're being consistent the American spelling is "behavior"

File: docs/tutorial_subproc_strings.rst [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+.. code-block:: xonsh
+
+    $ echo 'foo"bar"baz'
+    foo"bar"baz
+
+    $ echo '--key="value"'
+    --key="value"
+
+This is yet-another major point of departure for xonsh from traditional
```
Comment (gforsyth) [inline docs/tutorial_subproc_strings.rst]: No hyphen

File: docs/tutorial_subproc_strings.rst (hunk 3 above)
Comment (scopatz) [reply docs/tutorial_subproc_strings.rst]: Let's assume I am moving to Canada :smile: :canada:

**Comment by scopatz** [comment]
Thanks for the read through @gforsyth! Changes made.

**Comment by scopatz** [comment]
Also, this is what I get for writing docs at 2AM

**Comment by gforsyth** [comment]
Thanks, @scopatz!  We all benefit from you writing docs at 2am.

File: docs/tutorial_subproc_strings.rst [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+to try this in xonsh, you'd see:
+
+**xonsh**
+
+.. code-block:: xonsh
+
+    $ echo Actually\ Three\ Arguments
+    Actually\ Three\ Arguments
+
+In this example, echo recives three arguments:: ``"Actually\\"``, ``"Three\\"``,
```
Comment (jcrben) [inline docs/tutorial_subproc_strings.rst:40]: @scopatz I'm sure this is due to an obvious feature of Python which is not immediately clear to, but perhaps a short note explaining why echo actually receives two backslashes here?

also recives is a misspelling

Comment (jcrben) [reply docs/tutorial_subproc_strings.rst:40]: @anki-code I could make a pr if I understood why 🙂

**Pull request merged**